### ISSUE TITLE
Pkg.build error on MacOS

### ISSUE THREAD

**Issue by mkborregaard** [JuliaIO/ImageMagick.jl#66]
```
julia> Pkg.build("ImageMagick")
INFO: Building ImageMagick
=============================[ ERROR: ImageMagick ]=============================

LoadError: MethodError: no method matching generate_steps(::BinDeps.LibraryDependency, ::BinDeps.CustomPathBinaries, ::Dict{Symbol,Any})
Closest candidates are:
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.AptGet, ::Any) at /Users/michael/.julia/v0.5/BinDeps/src/dependencies.jl:320
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.Yum, ::Any) at /Users/michael/.julia/v0.5/BinDeps/src/dependencies.jl:331
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.Pacman, ::Any) at /Users/michael/.julia/v0.5/BinDeps/src/dependencies.jl:343
  ...
while loading /Users/michael/.julia/v0.5/ImageMagick/deps/build.jl, in expression starting on line 110

================================================================================
ERROR: UndefVarError: BinDeps not defined
 in deserialize(::SerializationState{IOStream}, ::Type{Module}) at ./serialize.jl:602
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in deserialize(::SerializationState{IOStream}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in deserialize_datatype(::SerializationState{IOStream}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in deserialize(::SerializationState{IOStream}, ::DataType) at ./serialize.jl:881
 in deserialize_datatype(::SerializationState{IOStream}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in deserialize(::SerializationState{IOStream}, ::DataType) at ./serialize.jl:881
 in deserialize_datatype(::SerializationState{IOStream}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in handle_deserialize(::SerializationState{IOStream}, ::Int32) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in (::Base.Pkg.Entry.##54#55{Dict{Any,Any}})(::IOStream) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in open(::Base.Pkg.Entry.##54#55{Dict{Any,Any}}, ::String, ::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in build!(::Array{String,1}, ::Dict{Any,Any}, ::Set{Any}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in build(::Array{String,1}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,VersionNumber}, ::Set{String}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}) at ./pkg/entry.jl:476
 in edit(::Function, ::String, ::Base.Pkg.Types.VersionSet, ::Vararg{Base.Pkg.Types.VersionSet,N}) at ./pkg/entry.jl:30
 in (::Base.Pkg.Entry.##2#5{String,Base.Pkg.Types.VersionSet})() at ./task.jl:360
 in sync_end() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in macro expansion at ./task.jl:327 [inlined]
 in add(::String, ::Base.Pkg.Types.VersionSet) at ./pkg/entry.jl:51
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#add,Tuple{String}})() at ./pkg/dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#add,Tuple{String}}, ::String) at ./file.jl:59
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N}) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in add(::String) at ./pkg/pkg.jl:100
```

```
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753 (2016-09-19 18:14 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, broadwell)

julia> Pkg.status.(["ImageMagick", "Homebrew", "BinDeps"])
 - ImageMagick                   0.2.0
 - Homebrew                      0.4.2
 - BinDeps                          0.4.5
```

**Comment by bjarthur** [comment]
dup of https://github.com/JuliaIO/ImageMagick.jl/issues/65

**Comment by mkborregaard** [comment]
Does that mean that the suggested fix is to remove all things ImageMagick from the 0.5 folder and reinstall?

**Comment by bjarthur** [comment]
give it a try and let us know if it works!  particularly informative would be what version of imagemagick is installed in Homebrew/usr/deps/Cellar.

there was a period of a week or so during which homebrew upgraded imagemagick to 7.0.4, which is not compatible with `ImageMagick.jl`.  i believe that folks who did a `Pkg.update()` during this time might be the ones having problems like yours.

**Comment by mkborregaard** [comment]
That was it, yes, I `Pkg.update()` once a week (I am sure there isn't really any reason to), and I had 7.0.4.6. I removed the package, deleted two folders named "ImageMagick" from Homebrew/deps/usr/Cellar, and reinstalled, and now it works. Should I close or do you want it open for reference?

**Comment by bjarthur** [comment]
let's keep it open for awhile to make it more noticeable for others who might have the same problem.  a small set though as you must 1. use ImageMagick on OSX instead of QuartzImageIO, 2. have checked out out the master branch, and 3. done an update within a narrow timeframe.

there might be a fix we could implement using the `validate` keyword arg of `library_dependency` in `BinDeps`, but i don't have the time right now.  see line 13 of deps/build.jl if you want to take a stab at it.

**Comment by mkborregaard** [comment]
Cool. I use QuartzImageIO, though, but have both packages installed as recommended in the Readme of QuartzImageIO. And am not on master of ImageMagick, though I may have been shortly in the course of some troubleshooting in the past.

**Comment by timholy** [comment]
Presumably closed by #90. Feel free to reopen or comment if not.
